ample/modelling/rosetta_model.py

```diff
@@ -289,14 +307,126 @@ def cmd_add_restraints(self, cmd):
+        if self.multimer_modelling == multimer_definitions.DIMER:
+            nchains = 2
+        elif self.multimer_modelling == multimer_definitions.TRIMER:
+            nchains = 3
+        elif self.multimer_modelling == multimer_definitions.TETRAMER:
+            nchains = 4
+
+        restraint_str = energy_functions.RosettaFunctionConstructs().FLAT_HARMONIC
+        domain_termini_distance = self.sequence_length * 1.5
+        import string 
+        import itertools
+        chains = list(string.ascii_uppercase)[:nchains]
+        fstr = ""
+        for chain1, chain2 in itertools.combinations(chains, 2):
+            # First add the terminal restraints
+            optd = { 'atom1': 'CA',
+                     'res1_seq' : 1,
+                     'atom2': 'CA',
+                     'res2_seq' : self.sequence_length,
+                     'x0' : domain_termini_distance,
+                     'stddev' : 3.0,
+                     'tol' : 5.0 }
+            fstr += restraint_str.format(**optd) + os.linesep
+            # Now add the restraints between chains
+            for resseq in range(1, self.sequence_length + 1):
+                resseq = str(resseq)
+                optd = { 'atom1': 'CA',
+                        'res1_seq' : resseq + chain1,
+                        'atom2': 'CA',
+                        'res2_seq' : resseq + chain2,
+                        'x0' : 10,
+                        'stddev' : 3.0,
+                        'tol' : 5.0 }
+                fstr += restraint_str.format(**optd) + os.linesep   
+        with open(restraints_file, 'w') as w:
+            w.write(fstr)
+        return restraints_file
+    
+    def create_multimer_flagsfile(self,
+                                  broker_file=None,
+                                  symmetry_file=None,
+                                  constraints_file=None):
+        flagsfile = os.path.join(self.work_dir, 'multimer_flagsfile.txt')
+        optd = { 'broker_file' : broker_file,
+                'symdef_file' : symmetry_file,
+                'constraint_file' : constraints_file,
+                'fasta_file' : self.fasta,
+                'frags3' : self.frags_3mers,
+                'frags9' : self.frags_9mers }
+        with open(flagsfile, 'w') as w:
+            w.write(multimer_definitions.FLAGSFILE_STR.format(**optd))
+        return flagsfile
 
     def find_binary(self, name):
         """
         Find a rosetta binary on different platforms
         separate from object as it's currently used by the NMR stuff - which is in dire need of refactoring.
 
         """
-        assert self.rosetta_bin and os.path.isdir(self.rosetta_bin)
+        if not self.rosetta_bin and os.path.isdir(self.rosetta_bin):
+            raise RuntimeError("Cannot find rosetta_bin directory: {}".format(self.rosetta_bin))
         binaries = glob.glob(os.path.join(self.rosetta_bin, "{0}.*".format(name)))
         if not len(binaries):
```

Comment:
```
if len(binaries):
    ...
else:
    return False
```

Reply:
I tried this and it add extra lines of code and to me makes it harder to read. This way, it's obvious that with no binaries you return False at this point.